### DISCUSSION TITLE
What is the state of the project since release 2.9 ?

### DISCUSSION THREAD

**Discussion by lthoulon-locala** [square/retrofit#3753] [Q&A]
Hello. 

I see that release 2.9.0 was done in 2020. Since then there has been no release. I was wondering what was the global state of the project and if we would be seing more updates in the future ?

Thanks

**Comment by JakeWharton** [comment]
Is there a specific issue you are having? The project is fairly stable.

**Reply by lthoulon-locala** [reply]
No specific issue no, it just seemed that 2 years was a lot and i've seen some commits since then that didn't go into any release (didn( really dig into what they were I must say). I was just curious about the state and if there are any plans forward for this lib. 
I know OkHttp is now in version 4.10 for instance and a version 5 is in alpha. Is there any plans to update the dependency ?
Also is there any plans for new adapters ? I have reactor on top of my mind for instance.

Anyway, no issue, juste curious :)

**Reply by JakeWharton** [reply]
OkHttp 4.x (and 5.x) are written in Kotlin and we are not going to force that dependency on our consumers unless we also rewrite Retrofit in Kotlin. The 4.x and 5.x releases are binary compatible with 3.x so you are able to bump the dependency yourself if you want.

Reactor adapter is here: https://github.com/JakeWharton/retrofit2-reactor-adapter. It's not in the core library because from what I remember they allow breaking API changes without major version bumps which is not something we can risk as a first-party adapter. Also the whole adapter pattern is meant so that anyone can write adapters for their library of choice. We don't aim to provide every one that is possible because it places too great a maintenance burden on us.

In general, the library is mostly stable and doesn't require a ton of work, but also I don't have a lot of time to maintain it right now. Hopefully in a few months my focus at work will allow bringing back in some of our open source libraries into weekly maintenance and forward progress. But definitely not sooner than September.

**Reply by chatko** [reply]
@JakeWharton 
first, I appreciate your work/time on this project :) 

I was looking for a release to get us gson 2.9.0 (due to CVEs). Currently doing an explicit exclude and import of gson 2.9.0 in my project to override. I saw your comment  https://github.com/square/retrofit/issues/3691#issuecomment-1035088817 and I think that patching (eg excluding and updating) transitive dependencies can be risky and makes using the project more difficult. 

 any ways I hope we'll still see a release this year.

**Reply by JakeWharton** [reply]
You never need to explicitly exclude transitive dependencies if your goal is only to use a newer version. You only need to declare the new dependency in your module and the new version will be resolved. All major build systems that feature dependency resolution will ensure that the newer version is resolved and that only a single jar is put on the classpath.

I expect there to be a release by the end of the year. The other projects which have stolen much of the attention of those who maintain this project (and a few others) are slowing down to only require part of our time rather than 100% of it.

**Comment by BjoernAkAManf** [comment]
Just a note on why a new Release might be a good idea:

Running under Java 16 with JPMS enabled will cause issues. The corresponding PR #3618 does fix that, but requires usage of Snapshots.

e.g. 

```
    <repositories>
        <repository>
            <id>SonatypeOSS-s1</id>
            <name>Sonatype Snapshots</name>
            <url>https://s01.oss.sonatype.org/content/repositories/snapshots/</url>
            <releases>
                <enabled>false</enabled>
            </releases>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </repository>
    </repositories>

    <dependencies>
 [...]
        <dependency>
            <groupId>com.squareup.retrofit2</groupId>
            <artifactId>retrofit</artifactId>
            <version>2.10.0-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>com.squareup.retrofit2</groupId>
            <artifactId>converter-gson</artifactId>
            <version>2.10.0-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>com.squareup.retrofit2</groupId>
            <artifactId>converter-scalars</artifactId>
            <version>2.10.0-SNAPSHOT</version>
        </dependency>
    </dependencies>
```

Issue is not grave, but it did cost me about half an hour to figure out completely.

Leaving this here so that Information may help someone in the Future.

**Comment by guillaume-tgl** [comment]
Another argument for a new release would to provide updated R8 rules 😊

**Reply by minas1** [reply]
This. I'm also waiting for the updated R8 rules.
